src/components/header/gist-modal/renderer.tsx

```diff
@@ -215,17 +215,20 @@ class GistModal extends React.PureComponent<Props, State> {
               invalidFilename: true
             });
             return Promise.reject('Invalid file name');
-          } else {
-            const {revision, filename} = this.state.gist;
-            JSON.parse(json.files[filename].content);
-            if (this.state.latestRevision) {
-              this.props.history.push(`/gist/${gistId}/${filename}`);
-            } else {
-              this.props.history.push(`/gist/${gistId}/${revision}/${filename}`);
-            }
-            closePortal();
           }
+
+          return fetch(json.files[this.state.gist.filename].raw_url); // fetch from raw_url to handle large files
+        }
+      })
+      .then(async res => {
```

Comment:
Let's use await instead of then.

Reply:
Sure, i just wanted to stick to the code style there :)

Reply:
Thanks for looking out for the code style. I like to use every opportunity for refactoring.

Reply:
Changed that bit